lib/permutations.rb

```diff
@@ -1,4 +1,44 @@
-
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+  return false if string1.length != string2.length
+
+  hash_table = generate_hash_table(string1.chars)
+
+  string2.chars.each do |char|
+    index = to_ascii(char) % string1.length
+    if hash_table[index].empty?
+      return false
+    else
+      hash_table[index] << char
+    end
+  end
+
+  #if any bucket has an uneven number of elements, then not a palindrome
+  if hash_table.any? { |element| element.length % 2 == 1 }
```

Comment:
Appending the char to the array on line 11 causes the failure that I mentioned in my comments above ('aaab' and 'baaa') here on line 16.  If both strings have an odd number of the same character, the test here will detect an even number and pass the test.